File: 205-Isomorphic-Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if not len(s) == len(t):
            return False
        isomorph = dict()
        for index, val in enumerate(s):
            if val in isomorph.keys():
                #print(val, " --> ", isomorph[val], "\t", t[index])
                if not isomorph[val] == t[index]:
                    return False
            else:
                if t[index] in isomorph.values():
                    return False
                isomorph[val] = t[index]
                
                #print(val, " ---> ", isomorph[val], "\t", t[index])
        return True

File: 205-Isomorphic-Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int m1[128], m2[128];
        for(int i = 0; i < 128; i++){
            m1[i] = m2[i] =  -1;
        }
        for(int i = 0; i < s.size(); i++){
            if(m1[s[i]] != m2[t[i]]){
                return false;
            }
            m1[s[i]] = i;
            m2[t[i]] = i;
        }
        return true;
    }
};
